docs/topics/kubernetes-developers.md

```diff
@@ -1,76 +1,124 @@
+    ...
 }
 ```
-NOTE: Make sure the service principal provided to run `aks-engine deploy` has access to pull images from this private registry. https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal#use-an-existing-service-principal
 
 * AKS Engine defaults to the `ubuntu` "distro" for `customHyperkubeImage` scenarios, which will build Linux VM nodes using the Ubuntu 16.04-LTS image SKU. You may also use Ubuntu 18.04-LTS-built VMs by explicitly setting the "distro" configuration accordingly. The "distro" configuration is applied distinctly to master VMs, and to the VMs in any configured agent pools (you may use different "distro" values in the masterProfile, and in each agentPoolProfile, according to your cluster requirements), for example:
 
 ```
 ...
 "masterProfile": {
     ...
-    "distro": "ubuntu-18.04"
+    "distro": "ubuntu-18.04",
     ...
 },
 ...
 "agentPoolProfiles": [
     {
         ...
-        "distro": "ubuntu-18.04"
+        "distro": "ubuntu-18.04",
         ...
     }
 ]
 ...
 ```
 
 * Run `aks-engine deploy` [as normal](../tutorials/deploy.md).
+
+## Kubernetes 1.17+
+
+Since Kubernetes has removed `make WHAT=cmd/hyperkube` command and AKS Engine has stopped supporting hyperkube for Kubernetes 1.17+, we will have to build individual Kubernetes components and deploy them separately. That includes [kube-apiserver](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/), [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), [kube-scheduler](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/), [kube-proxy](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/), [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/), and [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/).
+
+* Run the following commands to build Kubernetes and push images of individual components:
+
+```bash
+export KUBE_DOCKER_REGISTRY=<your-docker-registry>
+export KUBE_DOCKER_IMAGE_TAG=<your-custom-tag>
+make quick-release
+docker push <your-docker-registry>/kube-apiserver-amd64:<your-custom-tag>
+docker push <your-docker-registry>/kube-controller-manager-amd64:<your-custom-tag>
+docker push <your-docker-registry>/kube-proxy-amd64:<your-custom-tag>
+docker push <your-docker-registry>/kube-scheduler-amd64:<your-custom-tag>
+```
+
+After that, you will have to upload a tarball (`_output/release-tars/kubernetes-node-linux-amd64.tar.gz`) built from the previous step to online storage such as GitHub or Azure Storage. AKS Engine uses the tarball to extract kubelet and kubectl.
+
+### AKS Engine API Model
+
+* Open the AKS Engine API Model (e.g. a file from the examples directory) and add the following to the `orchestratorProfile` section:
+
+```
+"kubernetesConfig": {

```

Comment:
Do we want to mention that these workflows should also include specifying `"orchestratorRelease": "1.17"` in the api model? I think that will be required to properly instruct the template generation flow that this cluster configuration input should be interpreted in a 1.17 context.

Also, we might just mention this generally; when using custom images (hyperkube or not) you want to tell aks-engine which minor version (via `"orchestratorRelease"`) context you're in, so various defaults and configuration are properly applied to the ARM template + artifacts.